Allow to agree to new terms if already agreed to old terms

AntonKhorev opened the pull request:
In case *Contributor Terms* or *Terms of Use* were updated since last agreeing to them, which is supposedly the case for many people, it's impossible to accept the new terms.

This PR makes it possible by
- telling when the terms were accepted in *My Account*
- linking to the *Terms* page
  ![image](https://github.com/user-attachments/assets/3a64e797-ef17-4d1f-9c5d-a4b689b8299c)
- disabling the checks if terms were previously accepted on the *Terms* page, which made it do nothing or redirect back to *My Account*

pnorman commented:
> In case _Contributor Terms_ or _Terms of Use_ were updated since last agreeing to them

The Contributor Terms and Terms of use are different. The Contributor Terms are what grants the OSMF the right to distribute your data and they've never been updated. The [Terms of Use](https://osmfoundation.org/wiki/Terms_of_Use) cover accessing OSMF services. These have had minor updates, but they're not integrated into the website as they apply to all OSMF services, regardless of if the user has had an account.

Handling new CTs would be nice to have, but it's not a feature that would be used right now.

AntonKhorev commented:
> but they're not integrated into the website

https://github.com/openstreetmap/openstreetmap-website/pull/2028

AntonKhorev commented:
Since *terms* aren't part of the form, I moved them below the form. I also added *Terms of Use* label/section, which I didn't do previously because there's only one *terms* page.

AntonKhorev commented:
> There's no need to "accept" changes to the terms though, or rather the terms says that you're expected to check them when you want and that [continuing to use the service constitutes acceptance of any changes](https://osmfoundation.org/wiki/Terms_of_Use#XI._Changes_to_Services_or_Terms).

This interpretation contradicts some of changes in #2028. It added a db column named `tou_agreed`, which you don't need because it's impossible not to agree. Also it contradicts https://wiki.openstreetmap.org/wiki/GDPR/Affected_Services. Particularly the table at the beginning of that page has a *not accepted/unknown* column, which is impossible if the terms are accepted automatically.

AntonKhorev commented:
Do we just change the language from "accept" the ToU to "confirm that you read", or did we go from *mandatory* acceptance to *automatic* acceptance since https://github.com/openstreetmap/openstreetmap-website/pull/2028#issuecomment-439124390? With automatic acceptance we don't need to make any GDPR-related changes because whenever you use the data you "accept" the ToU and get all of the associated "rights". But obviously that's not what those who want GDPR-changes think. EWG, for example, asked me recently if I plan to "make accepting the terms of use a requirement". They think the ToU need to be "accepted". Now it may be the case that what EWG thinks is not the same what the current LWG thinks about all this.

Alternatively we can ignore all of this, forget about the ToU and use the current Privacy policy as a guide, see https://github.com/openstreetmap/openstreetmap-website/issues/5804#issuecomment-2726403892 minus the github argument. Show some things only to logged in users and ignore the `tou_agreed` attribute. Then this pull request is not very useful and so is #2028 from 2019, aside from adding links to the terms pages.

AntonKhorev commented:
> Just to confirm, does this PR result in actively presenting pre-2019 users with some sort of notice about the new terms, without having to visit the account settings first?

No it doesn't. There's no point in *actively presenting* something we can't agree to have for ~5 months. If/when we finally allow all users to visit the terms page, we can think about *actively presenting* it.

tomhughes commented on test/controllers/accounts/terms_controller_test.rb:
```diff
@@ -48,32 +48,72 @@ def test_show_not_seen_with_referer
+      end
+    end
+
+    def test_update_previously_accepted
```
We should probably have `test_decline_previously_accepted` to check that the dates don't change if you decline when revisiting the terms.

tomhughes reviewed (approved):
Thanks for the extra tests. I think this looks good now.
